### PR TITLE
Add zsh completions for #118 (credit @Mange)

### LINKED ISSUE #118: ZSH completion script
I wrote a ZSH completion script for myself, but I want to contribute it in case you want to include it for other users.

It's somewhat more intelligent than normal `clap` completion generated files because it understands the "command" argument and autocompletes other commands.

## Demo

[asciicast](https://asciinema.org/a/241277)

## Installation

Install script as `_watchexec` in your `fpath`.

```zsh
cd "${fpath[1]}"
```

Make it executable, and restart your shell. You can also load it at runtime without restarting by running `autoload -U _watchexec`.

## Script

```zsh
#compdef watchexec

setopt localoptions extended_glob

local cmd
local -a args
local -a _comp_priv_prefix

cmd="$words[1]"

args=(
  '(-c --clear)'{-c,--clear}'[Clear screen before executing command]'
  '(-h --help)'{-h,--help}'[Prints help information]'
  '(-k --kill)'{-k,--kill}'[Send SIGKILL to child processes (deprecated, use -s SIGKILL instead)]'
  '(-n --no-shell)'{-n,--no-shell}'[Do not wrap command in ''sh -c'' resp. ''cmd.exe /C'']'
  '(-p --postpone)'{-p,--postpone}'[Wait until first change to execute command]'
  '(-r --restart)'{-r,--restart}'[Restart the process if it''s still running]'
  '(-V --version)'{-V,--version}'[Prints version information]'
  '(-v --verbose)'{-v,--verbose}'[Print debugging messages to stderr]'
  '(-d --debounce)'{-d+,--debounce=}'[Set the timeout between detected change and command execution, defaults to 500ms]:milliseconds'
  '(-e --exts)'{-e+,--exts=}'[Comma-separated list of file extensions to watch (js,css,html)]:extensions'
  '(-f --filter)'{-f+,--filter=}'[Ignore all modifications except those matching the pattern]:pattern'
  '(-i --ignore)'{-i+,--ignore=}'[Ignore modifications to paths matching the pattern]:pattern'
  '(-w --watch)'{-w+,--watch=}'[Watch a specific directory]:path:_path_files -/'
  '(-s --signal)'{-s+,--signal=}'[Send signal to process upon changes, e.g. SIGHUP]:signal'
  '--force-poll=[Forces polling mode]:interval'
  '--no-default-ignore[Skip auto-ignoring of commonly ignored globs]'
  '--no-vcs-ignore[Skip auto-loading of .gitignore files for filtering]'
  '(-)1:command: _command_names -e'
  '*::arguments:{ _comp_priv_prefix=( $cmd -n ${(kv)opt_args[-u]} ) ; _normal }'
)

_arguments -s -S $args
```

### PR DESCRIPTION
**Pull request by svenstaro**
Fixes #118

### CODE REVIEW

**Comment by svenstaro** [comment]
If this is merged I propose the deletion of https://github.com/watchexec/watchexec/wiki/ZSH-completion

**Comment by passcod** [comment]
Can you add this to the packaging script so it gets included in the release {tar,zip}balls?

**Comment by passcod** [comment]
Also could you add the `--no-ignore` flag from this while we're at it? https://github.com/watchexec/watchexec/commit/e90bbcb9bde0efd8315a4844eadd914036af7123 Then I'll prep and release all this at once

**Comment by svenstaro** [comment]
@passcod Done for everything but Windows and Mac. I don't know how to do this with Homebrew and Windows probably doesn't need this.

File: ci/before_deploy.sh [hunk 1]
```diff
@@ -25,6 +25,7 @@ if [[ "$TARGET" == "x86_64-unknown-linux-gnu" ]]; then
     mkdir -p DEBIAN usr/bin usr/share/man/man1
     cp "../../$name/watchexec" usr/bin/
     cp "../../$name/watchexec.1" usr/share/man/man1/
+    cp "../../$name/zsh" usr/share/zsh/site-functions/_watchexec
```
Comment (passcod) [inline ci/before_deploy.sh]: You actually need to create the dir (see line 25) before copying into it :)

**Comment by svenstaro** [comment]
Should be fixed :)

**Comment by svenstaro** [comment]
@passcod Could you make a release with this?

**Pull request merged**